src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java

```diff
@@ -33,18 +37,27 @@ protected Processor(
         this.errorWrapper = errorWrapper;
     }
 
+    protected abstract ServiceBusHelper serviceBusHelper();
+
     protected void processParsedEnvelopeDocuments(
         Envelope envelope,
         List<Pdf> pdfs,
-        CloudBlockBlob cloudBlockBlob
+        CloudBlockBlob cloudBlockBlob,
+        ServiceBusHelper serviceBusHelper
     ) {
         if (!uploadParsedEnvelopeDocuments(envelope, pdfs)) {
             return;
         }
         if (!markAsUploaded(envelope)) {
             return;
         }
-        markAsProcessed(envelope);
+        if (!markAsProcessed(envelope)) {
+            return;
+        }
+        if (!sendProcessedMessage(serviceBusHelper, envelope)) {
+            return;
+        }
+        markAsNotified(envelope);
```

Comment:
`markAsProcessed` is wrapped, this is not.
Is this by design?

Reply:
They are all wrapped by handleEvent() for what I can see.

Reply:
It is definitely wrapped.